lib/puppet/util/windows/monkey_patches/dir.rb

```diff
@@ -0,0 +1,40 @@
+require 'win32/dir/constants'
+require 'win32/dir/functions'
+require 'win32/dir/structs'
+
+class DirMonkeyPatched
+  include ::Dir::Structs
+  include ::Dir::Constants
+  extend  ::Dir::Functions
+
+  path  = nil
+  key   = 'PERSONAL'
+  value = 0x0005
+  buf   = 0.chr * 1024
+  buf.encode!('UTF-16LE')
+
+  if SHGetFolderPathW(0, value, 0, 0, buf) == 0 # Current path
+    path = buf.strip
+  elsif SHGetFolderPathW(0, value, 0, 1, buf) == 0 # Default path
+    path = buf.strip
+  else
+    ptr   = FFI::MemoryPointer.new(:long)
```

Comment:
Can we use the block format of `MemoryPointer.new`  here? Mostly for readability, should be an easy change.

`FFI::MemoryPointer.new(:long) do |ptr| ...`